src/System.Composition.Hosting/src/Microsoft/Internal/Assumes.InternalErrorException.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.Runtime.Serialization;
+
+namespace System.Composition.Hosting
+{
+    internal partial class Assumes
+    {
+        // The exception that is thrown when an internal assumption failed.
+        [SuppressMessage("Microsoft.Design", "CA1064:ExceptionsShouldBePublic")]
+        private class InternalErrorException : Exception
```

Comment:
This and the related Assumes/Requires/etc. code is being duplicated across projects.  Can we not define them once and share those files, as we do with other Common code?